File: dsu.cpp

class DSU{
    public:
        DSU(int n){
            this->n=n;
            this->parent.resize(n);
            this->sz.resize(n);
        }
        void make_set(int v){
            parent[v]=v;
            sz[v]=1;
        }

        int find_set(int v){
            if(v==parent[v]){
                return v;
            }
            parent[v]=find_set(parent[v]);
            return parent[v];
        }

        void union_set(int a,int b){
            a=find_set(a);
            b=find_set(b);
            if(a==b){
                return;
            }
            if(sz[a]<sz[b]){
                swap(a,b);
            }
            parent[b]=a;
            sz[a]+=sz[b];
        }
        


    private:
        vector<int>parent;
        vector<int>sz;
        int n;
};

File: dsu.py
class dsu(): #ta
	def __init__(self,n):
		self.parent=[0]*(n)
		self.sz=[0]*(n)

	def make_set(self,v):
		self.parent[v]=v
		self.sz[v]=1

	def find_set(self,v):
		if v==self.parent[v]:
			return v
		self.parent[v]=self.find_set(self.parent[v])
		return self.parent[v]

	def union(self,a,b):
		a=self.find_set(a)
		b=self.find_set(b)
		if a==b:
			return
		if self.sz[a]<self.sz[b]:
			a,b=b,a
		self.parent[b]=a
		self.sz[a]+=self.sz[b]

	def getsize(self,v):
		return self.sz[self.find_set(v)]
